src/System.Net.Http/src/System/Net/Http/Managed/HttpConnectionKey.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Net.Http
+{
+    internal struct HttpConnectionKey : IEquatable<HttpConnectionKey>
+    {
+        public readonly bool UsingSSL;
+        public readonly string Host;
+        public readonly int Port;
+
+        public HttpConnectionKey(Uri uri)
+        {
+            if (uri.Scheme == "http")
+            {
+                UsingSSL = false;
+            }
+            else if (uri.Scheme == "https")
+            {
+                UsingSSL = true;
+            }
+            else
+            {
+                throw new ArgumentException("Invalid Uri scheme", nameof(uri));
+            }
+
+            Host = uri.Host;
+            Port = uri.Port;
+        }
+
+        public override int GetHashCode()
+        {
+            return UsingSSL.GetHashCode() ^ Host.GetHashCode() ^ Port.GetHashCode();
```

Comment:
Thanks, Morgan.  Good feedback, but I'm going to leave it as-is for the purposes of this PR.  @geoffkizer or anyone else (you? :smile:) can address this post initial check-in.  There's a lot of stuff to improve.